gchart-parent/gchart/src/main/java/org/wicketstuff/gchart/Chart.java

```diff
@@ -0,0 +1,381 @@
+import org.wicketstuff.gchart.gchart.options.ChartOptions;
+
+/**
+ * Abstraction of Google charts for wicket. {@code OutputMarkupId} is set to
+ * true, since id is referenced in generated JavaScript.
+ *
+ * @author Dieter Tremel
+ */
+public class Chart extends WebComponent implements IClusterable, JavaScriptable {
+
+    private static final long serialVersionUID = 1L;
+    private static final Logger log = LoggerFactory.getLogger(Chart.class);
+    /** URL for Google lib loader */
+    public static final String LOADER_URL = "https://www.gstatic.com/charts/loader.js";
+
+    private ChartLibLoader loader = null;
+    private boolean responsive = true;
+    private Locale locale = null;
+    private String mapsApiKey = null;
+    private IModel<ChartType> typeModel;
+    private IModel<DataTable> dataModel;
+
+    /**
+     * Basic Constructor.
+     *
+     * @param id Wicket id. Id is used in javascript, but at the moment not
+     * escaped to JavaScript identifier rules. So use only characters allowed
+     * for JavaScript declarations to avoid problems.
+     */
+    public Chart(String id) {
+        super(id);
+        setOutputMarkupId(true);
+    }
+
+    /**
+     * Complete Constructor.
+     *
+     * @param id Wicket id. Id is used in javascript, but at the moment not
+     * escaped to JavaScript identifier rules. So use only characters allowed
+     * for JavaScript declarations to avoid problems.
+     * @param typeModel Model of Type of chart.
+     * @param optionModel Model of optionModel.
+     * @param dataModel Model of data dataModel.
+     */
+    public Chart(String id, IModel<ChartType> typeModel, IModel<ChartOptions> optionModel, IModel<DataTable> dataModel) {
+        this(id);
+        this.typeModel = typeModel;
+        this.setDefaultModel(optionModel);
+        setDataModel(dataModel);
+//        add(new GoogleChartBehavior(this, getMarkupId()));
+    }
+
+    /**
+     * Complete Constructor.
+     *
+     * @param id Wicket id. Id is used in javascript, but at the moment not
+     * escaped to JavaScript identifier rules. So use only characters allowed
+     * for JavaScript declarations to avoid problems.
+     * @param typeModel Model of Type of chart.
+     * @param optionModel Model of optionModel.
+     * @param dataModel Model of data dataModel.
+     * @param loader Loader to add the chart to.
+     */
+    public Chart(String id, IModel<ChartType> typeModel, IModel<ChartOptions> optionModel, IModel<DataTable> dataModel, ChartLibLoader loader) {
+        this(id);
+        this.typeModel = typeModel;
+        this.setDefaultModel(optionModel);
+        setDataModel(dataModel);
+//        add(new GoogleChartBehavior(this, getMarkupId()));
+        this.loader = loader;
+        loader.addChart(this);
+    }
+
+    /**
+     * Create the JavaScript for the Google loader.
+     * @return Header Item with Google 
+     */
+    public static JavaScriptHeaderItem cerateLoaderItem() {

```

Comment:
Does it have to be `static` ?
If it is not static then one could override it to use a different url in case Google ever changes it.